core/src/test/java/com/linecorp/armeria/client/HttpClientResponseTimeoutTest.java

```diff
@@ -117,6 +119,54 @@ void whenTimedOut() {
         });
     }
 
+    @Test
+    void timeoutWithContext() {
+        final WebClient client = WebClient
+                .builder(server.httpUri())
+                .build();
+        try (ClientRequestContextCaptor ctxCaptor = Clients.newContextCaptor()) {
+            final CompletableFuture<AggregatedHttpResponse> response = client.get("/no-timeout").aggregate();
+            final ClientRequestContext cctx = ctxCaptor.get();
+            cctx.timeoutNow();
+            assertThat(cctx.isTimedOut()).isFalse();
```

Comment:
This might be flaky. According to the CPU scheduling, `isTimedOut()` could be returned true.
How about checking the condition before calling `timeoutNow()`.